diff --git a/sim/arm/armsupp.c b/sim/arm/armsupp.c
index b8e1d38..1968021 100644
--- a/sim/arm/armsupp.c
+++ b/sim/arm/armsupp.c
@@ -627,7 +627,7 @@ ARMul_MCR (ARMul_State * state, ARMword instr, ARMword source)
 
 /* This function does the Busy-Waiting for an MRC instruction.  */
 
-ARMword
+ARMword 
 ARMul_MRC (ARMul_State * state, ARMword instr)
 {
   unsigned cpab;
@@ -636,7 +636,7 @@ ARMul_MRC (ARMul_State * state, ARMword instr)
   if (! CP_ACCESS_ALLOWED (state, CPNum))
     {
       ARMul_UndefInstr (state, instr);
-      return;
+      return (0);
     }
 
   cpab = (state->MRC[CPNum]) (state, ARMul_FIRST, instr, &result);
